bootstrap/config/config.go

```diff
@@ -511,7 +512,11 @@ func (cp *Processor) listenForChanges(serviceConfig interfaces.Configuration, co
 					lc.Info("Insecure Secrets have been updated")
 					secretProvider := container.SecretProviderFrom(cp.dic.Get)
 					if secretProvider != nil {
-						secretProvider.SecretsUpdated()
+						// Find the updated secret's path and perform call backs.
+						updatedSecrets := getMapDiff(previousInsecureSecrets, currentInsecureSecrets)
+						for _, v := range updatedSecrets {
+							secretProvider.SecretsUpdatedWithPath(v)
+						}
```

Comment:
Should we make sure this update is protected by mutex to have threading safe?

Reply:
Agree.  We don't want to diff the map if someone is in the middle of updating it.